MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -742,27 +751,12 @@ public final class MagazineLayout: UICollectionViewLayout {
       }
 
     case .decorationView:
-      contentOffsetAdjustment = nil
       assertionFailure("`MagazineLayout` does not support decoration views")
 
     @unknown default:
-      contentOffsetAdjustment = nil
       assertionFailure("`MagazineLayout` does not support this kind of element category")
     }
 
-    let context = super.invalidationContext(
-      forPreferredLayoutAttributes: preferredAttributes,
-      withOriginalAttributes: originalAttributes) as! MagazineLayoutInvalidationContext
-
-    if let contentOffsetAdjustment, !isPerformingBatchUpdates {
-      // If we're in the middle of a batch update, we need to adjust our content offset. Doing it
-      // here in the middle of a batch update gets ignored for some reason. Instead, we delay
-      // slightly and do it in `finalizeCollectionViewUpdates`.
-      context.contentOffsetAdjustment = contentOffsetAdjustment
-    }
-
-    context.invalidateLayoutMetrics = false
-
```

Comment:
I haven't found a need for this special handling after this refactor - another simplification!